src/globus_action_provider_tools/authentication.py

```diff
@@ -70,14 +71,17 @@ def __init__(
         auth_client: ConfidentialAppAuthClient,
         bearer_token: str,
         expected_scopes: frozenset[str],
+        client_factory: ClientFactory | None = None,
     ) -> None:
         self.auth_client = auth_client
         self.bearer_token = bearer_token
         self.sanitized_token = self.bearer_token[-7:]
         self.expected_scopes = expected_scopes
+        self._client_factory = (
+            client_factory if client_factory is not None else ClientFactory()
+        )
```

Comment:
Is this equivalent?

```suggestion
        self._client_factory = client_factory or ClientFactory()
```

Reply:
Yeah, I'm applying the change you suggested.

I've gotten bitten by truthiness checks vs `is None` in the past, and I still don't always trivially decide whether to use the implicit `bool()` logic or `is None` checking. I think one rule, which this case matches, is easy: it's fine to use implicit `bool()` when the non-None type always `bool()`s as `True`.